src/test/resources/libs/test_cross_class_usage/vars/monster1.groovy

```diff
@@ -0,0 +1,5 @@
+import org.test.Monster1
+
+void call(Monster1 m1) {
```

Comment:
What if you change `Monster1` to just an `Object`?

Reply:
The method interceptor matches the call. Here's the debug output.

```
Loading shared library test_cross_class_usage with version master

DEBUG_ARGTYPE GCL.loadClass source=file:/C:/src/JenkinsPipelineUnit/origin/vars-with-cross-class-args/build/resources/test/libs/test_cross_class_usage/src/org/test/Monster1.groovy
DEBUG_ARGTYPE GCL.loadClass cls=class org.test.Monster1
DEBUG_ARGTYPE GCL.loadClass cls.hashCode()=737985704

DEBUG_ARGTYPE PTH.setGlobalVars monster1.call.nativeParameterTypes=[class java.lang.Object]
DEBUG_ARGTYPE PTH.setGlobalVars nativeParameterTypes*.hashCode()=[1161667116]
DEBUG_ARGTYPE PTH.setGlobalVars cachedParamClasses=[java.lang.Object]
DEBUG_ARGTYPE PTH.setGlobalVars cachedParamClasses*.hashCode()=[1063877011]

DEBUG_ARGTYPE PTH.getAllowedMethodEntry argsClasses=[class org.test.Monster1]
DEBUG_ARGTYPE PTH.getAllowedMethodEntry argsClasses*.hashCode()=[737985704]
DEBUG_ARGTYPE PTH.getAllowedMethodEntry casted to paramTypes=[class org.test.Monster1]
DEBUG_ARGTYPE PTH.getAllowedMethodEntry casted to paramTypes*.hashCode()=[737985704]

DEBUG_ARGTYPE MethodSignature.equals thisClazz=class java.lang.Object; thatClazz=class org.test.Monster1
DEBUG_ARGTYPE MethodSignature.equals thisClazz.canonicalName=java.lang.Object; thatClazz.canonicalName=org.test.Monster1
DEBUG_ARGTYPE MethodSignature.equals thisClazz.hashCode()=1161667116; thatClazz.hashCode()=737985704
DEBUG_ARGTYPE MethodSignature.equals Signatures are compatible
Dracula makes quite a scary monster
````

And, as expected, the same thing happens if the type is `def` or ommitted.